File: 229hard/seven.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

int rev(int num)
{
    int rev = 0;
    while(num > 0)
    {
        rev *= 10;
        rev += num % 10;
        num /= 10;
    }

    return rev;
}

int main(int argc, char* argv[]) 
{
    int e = atoi(argv[1]);
    unsigned long long int result = 0;

    for(int i = 0; i < pow(10, e); i += 7)
    {
        if(rev(i) % 7 == 0)
        {
            result += i;
        }
    }

    printf("%llu\n", result);

    return 0;
}

File: 229hard/seven.py
def rev_int(number):
    rev = 0
    while number > 0:
        rev *= 10
        rev += number % 10
        number //= 10
    return rev


def mod_seven_sum(e):
    result = 0
    for i in range(0, 10**e, 7):
        if rev_int(i) % 7 == 0:
            result += i
    print(result)
